### DISCUSSION TITLE
I can't input right words in vim while I am using the x11 example to connect to a server,What should I do?

### DISCUSSION THREAD

**Discussion by Oceanboy-free** [libssh2/libssh2#1063] [Q&A]
**Describe the bug**
I can't input right words in vim while I am using the x11 example to connect to a server,What should I do?
**To Reproduce**
Steps to reproduce the behavior.

use the x11 in the example to test

A recipe or example code that reproduces the problem? A stack trace from a crash

**Expected behavior**
I want to edit files while I use libssh2 to connect to other server

**Version (please complete the following information):**
 - OS and version: [e.g. iOS, macOS, Windows, Linux (distro)] debian 11
 - libssh2 version: [e.g. 1.10.0] 1.10.0
 - crypto backend: [OpenSSL, MbedTLS, Libgcrypt, WinCNG, OS400, WolfSSL, None]
none
**Additional context**
![libssh2-example](https://user-images.githubusercontent.com/37373085/155451357-39a90186-30dc-46d0-8e3e-40ad7b280353.png)
![libssh2-vim](https://user-images.githubusercontent.com/37373085/155451366-5c4e4073-3003-45b2-974d-5e3f5d0645ef.png)

**Comment by dfandrich** [comment]
I don't think that example program will set the TERM variable on the remote
shell. Try setting that manually before running vim.

**Comment by Oceanboy-free** [comment]
I got the example in this folder,could you please tell me how could I chang this example,so that I can use the vim in the x11 example?
![image](https://user-images.githubusercontent.com/37373085/155473370-9c02a904-7be3-4656-9fc3-d3237899d66b.png)
![image](https://user-images.githubusercontent.com/37373085/155473481-2ebd876a-b1f4-4491-abb3-82f42f73ed5d.png)

**Comment by dfandrich** [comment]
You can test this theory without any code. Just log in to the server by ssh and
type "export TERM=whatever" and replace "whatever" with the proper terminal
type (probably xterm). Then start vim.

**Comment by Oceanboy-free** [comment]
I tried this but it seems the same.
![image](https://user-images.githubusercontent.com/37373085/155693561-afc5fe4c-f2b5-428e-8772-c775131de274.png)

![image](https://user-images.githubusercontent.com/37373085/155693140-1b598e55-9d63-495b-a01c-8080d90cca30.png)

And I found that the example may have set the TERM in its code,I found this code in the X11 example.

![image](https://user-images.githubusercontent.com/37373085/155693719-484f5bf5-77b2-4be1-9eef-1e7ec50c21dc.png)
